app/src/main/java/com/google/samples/apps/sunflower/compose/plantdetail/PlantDetailView.kt

```diff
@@ -0,0 +1,421 @@
+import dev.chrisbanes.accompanist.mdctheme.MaterialThemeFromMdcTheme
+
+
+/**
+ * As these callbacks are passed in through multiple Composables, to avoid having to name
+ * parameters to not mix them up, they're aggregated in this class.
+ */
+data class PlantDetailsCallbacks(
+    val onFabClicked: () -> Unit,
+    val onBackClicked: () -> Unit,
+    val onShareClicked: () -> Unit
+)
+
+@Composable
+fun PlantDetailsScreen(
+    plantId: String,
+    onBackClicked: () -> Unit,
+    onShareClicked: (String) -> Unit
+) {
+    // ViewModel and LiveDatas needed to populate the plant details info on the screen
+    val plantDetailsViewModel: PlantDetailViewModel = viewModel(
+        factory = InjectorUtils.providePlantDetailViewModelFactory(ContextAmbient.current, plantId)
+    )
+    val plant = plantDetailsViewModel.plant.observeAsState().value
+    val isPlanted = plantDetailsViewModel.isPlanted.observeAsState().value
+
+    // Every time there's a new value for plant or isPlanted LiveData, this block will get executed
+    if (plant != null && isPlanted != null) {
+        val context = ContextAmbient.current
+        Surface {
+            var showSnackbar by state { false }
+
+            TextSnackbarContainer(
+                snackbarText = stringResource(R.string.added_plant_to_garden),
+                showSnackbar = showSnackbar,
+                onDismissSnackbar = { showSnackbar = false }
+            ) {
+                PlantDetails(plant, isPlanted, PlantDetailsCallbacks(
+                    onBackClicked = onBackClicked,
+                    onFabClicked = {
+                        plantDetailsViewModel.addPlantToGarden()
+                        showSnackbar = true
+                    },
+                    onShareClicked = {
+                        val shareText = context.resources.getString(R.string.share_text_plant, plant.name)
+                        onShareClicked(shareText)
+                    }
+                ))
+            }
+        }
+    }
+}
+
+@VisibleForTesting
+@Composable
+fun PlantDetails(
+    plant: Plant,
+    isPlanted: Boolean,
+    callbacks: PlantDetailsCallbacks,
+    modifier: Modifier = Modifier
+) {
+    // PlantDetails owns the scrollerPosition to simulate CollapsingToolbarLayout's behavior
+    val scrollerPosition = ScrollerPosition()
+    var plantScroller by state { PlantDetailsScroller(scrollerPosition, 0f) }
+
+    // Transition that fades in/out the header with the image and the Toolbar
+    Transition(
+        definition = toolbarTransitionDefinition,
+        toState = plantScroller.toolbarState
+    ) { transitionState ->
+        Stack(modifier) {
+            PlantDetailsContent(
+                scrollerPosition = plantScroller.scrollerPosition,
+                toolbarShown = plantScroller.toolbarShown,
+                onNamePositioned = { newNamePosition ->
+                    if (plantScroller.namePosition == 0f) {
```

Comment:
Should this compare to `plantScroller.namePosition` instead of 0?

Reply:
Changed it to Float.MIN_VALUE instead and added a comment as I just want the original position of name on the screen (i.e. just want one update)